engine/src/main/java/org/hibernate/search/query/engine/impl/LuceneHSQuery.java

```diff
@@ -66,7 +65,7 @@
  * @author Emmanuel Bernard
  * @author Hardy Ferentschik
  */
-public class LuceneHSQuery extends AbstractHSQuery implements HSQuery, Serializable {
```

Comment:
yes I realize it's redundant, but wondering what is the contract you're needing.
In practice, it's `LuceneHSQuery` which you need to be serializable, right?

We could make it serializable without having to mandate the contract on its parent, or we might want to mark both as Serializable as an explicit marker (reminder).